core/domain/collection_services.py

```diff
@@ -555,28 +555,14 @@ def get_collection_ids_matching_query(query_string, cursor=None):
         collection_ids, search_cursor = search_services.search_collections(
             query_string, remaining_to_fetch, cursor=search_cursor)
 
-        invalid_collection_ids = []
-        for ind, model in enumerate(
+        for ind, _ in enumerate(
                 collection_models.CollectionSummaryModel.get_multi(
                     collection_ids)):
-            if model is not None:
-                returned_collection_ids.append(collection_ids[ind])
-            else:
-                invalid_collection_ids.append(collection_ids[ind])
+            returned_collection_ids.append(collection_ids[ind])
 
         if len(returned_collection_ids) == feconf.SEARCH_RESULTS_PAGE_SIZE or (
                 search_cursor is None):
             break
-        else:
```

Comment:
The number of collections in a page is always lesser or equal to max page size. Hence the two conditions below are unreachable.

Reply:
Done